### PR TITLE
Add font tarballs to repository

### PR DESCRIPTION
**Pull request by xjuan**
This avoids checksum errors everytime a font changes upstream.
Use fonts/download_fonts.sh to update fonts when desired.

https://phabricator.endlessm.com/T19905

### CODE REVIEW

**Comment by ebassi** [comment]
It would be neat if you could modify the script that generates the manifest to automatically get the SHA checksum of each font package and fill up the `sha256` field in the JSON file.

File: generate-manifest.py [hunk 1]
```diff
@@ -76,6 +77,30 @@ def edit_manifest(data, arch, branch, runtime_version):
...
+
+    for font in os.listdir('fonts'):
+        if font.endswith('.zip'):
+            path = 'fonts/' + font
+            sources.append({ 'type': 'file',
+                             'path': path,
+                             'dest-filename': font,
+                             'sha256': sha256(path)})
+
+    data['modules'].append({ 'name': 'default-theme-fonts',
```
Comment (ebassi) [inline generate-manifest.py]: I'd rather have `data['modules'].prepend()` here, for two reasons:

 - early failure
 - keep the `os-release` module at the end, as it's the last part of the build process

File: fonts/download_fonts.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash -e
+
+GOOGLE_FONTS=(Fira+Sans Lato Libre+Baskerville Merriweather Noto+Sans Noto+Serif Raleway Roboto)
```
Comment (ebassi) [inline fonts/download_fonts.sh]: Can we add this as a `Makefile` rule, instead?

File: generate-manifest.py [hunk 3]
```diff
@@ -76,6 +77,30 @@ def edit_manifest(data, arch, branch, runtime_version):
...
+    checksum = hashlib.sha256()
+    with open(filename, 'rb') as file:
+        for data in iter(lambda: file.read(65536), b''):
+            checksum.update(data)
+    return checksum.hexdigest()
+
+def add_fonts_module(data):
+    """Add fonts module to manifest json"""
+
+    sources = [ { 'type': 'file', 'path': 'fonts-makefile', 'dest-filename': 'Makefile' } ]
```
Comment (ebassi) [inline generate-manifest.py]: Since we're updating the manifest, I think we should also drop the `fonts-makefile` hack, and use `"build": "simple"`.

**Review by ebassi — CHANGES_REQUESTED** [review]

File: generate-manifest.py [hunk 4]
```diff
@@ -76,6 +77,34 @@ def edit_manifest(data, arch, branch, runtime_version):
             data['modules'].insert(0, gtk_module)
             break
 
+def sha256(filename):
+    checksum = hashlib.sha256()
+    with open(filename, 'rb') as file:
```
Comment (ptomato) [inline generate-manifest.py:82]: For next time: `file` is a built-in in Python, so it's better not to override it

**Review by ptomato — APPROVED** [review]
Looks good, one style nitpick but no need to block the pull request on it.

**Pull request merged**